pkg/service/service.go

```diff
@@ -1563,18 +1567,16 @@ func (s *Service) updateBackendsCacheLocked(svc *svcInfo, backends []*lb.Backend
 			if s.backendRefCount.Add(hash) {
 				id, err := AcquireBackendID(backend.L3n4Addr)
 				if err != nil {
+					s.backendRefCount.Delete(hash)
 					return nil, nil, nil, fmt.Errorf("Unable to acquire backend ID for %q: %s",
 						backend.L3n4Addr, err)
 				}
 				backends[i].ID = id
-				backends[i].Weight = backend.Weight
```

Comment:
How's this change related to the issue this commit is fixing?

Reply:
this one is really not related to anything, to be honest. I just noticed this LoC that does absolutely nothing, so I figured it's fine just adding it here.